examples/gno.land/r/demo/gnofundme/gnofundme.gno

```diff
@@ -0,0 +1,253 @@
+package gnofundme
+
+import (
+	"bytes"
+	"std"
+	"strconv"
+	"strings"
+	"time"
+
+	"gno.land/p/demo/avl"
+	pkg "gno.land/p/demo/gnofundme"
+	"gno.land/p/demo/ufmt"
+)
+
+var (
+	campaigns      *avl.Tree
+	users          *avl.Tree
+	campaignPrefix = "/r/demo/gnofundme:"
+)
+
+func init() {
+	campaigns = avl.NewTree()
+	users = avl.NewTree()
+}
+
+func NewUser(name string, email string) string {
+	// get the user address
+	address := std.GetOrigCaller()
+	// create the user
+	user := pkg.NewUser(name, email, address)
+
+	// the avl.tree key will be address to simplify the fetching
+	users.Set(address.String(), user)
+
+	return user.GetName() + " added successfully"
+}
+
+func NewCampaign(title string, description string, goal uint, beginT int64, deadlineT int64, beneficiary std.Address) (int, string) {
+	// get the caller as the owner address
+	caller := std.GetOrigCaller()
+	user := pkg.NewUser("user1", "email1", caller)
+
+	// the avl.tree key will be address to simplify the fetching
+	users.Set(caller.String(), user)
+
+	user = pkg.NewUser("user2", "email2", beneficiary)
+
+	// the avl.tree key will be address to simplify the fetching
+	users.Set(beneficiary.String(), user)
+	// verify that the user exists
+	o, ok := users.Get(caller.String())
+	if !ok {
+		panic("owner not found")
+	}
+
+	// verify that the benefeciary exists
+	b, ok := users.Get(beneficiary.String())
```

Comment:
It will exist because you created it in line 49, but you're doing the order in reverse. You should first check the user exists, and if it doesn't, create a user for that beneficiary. Same as with the owner.

Reply:
same.